resources/pre-upgrade.sh

```diff
@@ -5,11 +5,71 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+# shellcheck disable=SC1091
+source /util.sh
+
 NEXUS_DATA_DIR=/var/lib/nexus
+MIGRATION_HELPER_JAR="${NEXUS_DATA_DIR}/nexus-db-migrator-3.70.3-01.jar"
 
 FROM_VERSION="${1}"
 TO_VERSION="${2}"
 
 if [[ $FROM_VERSION == 2* ]] && [[ $TO_VERSION == 3* ]]; then
     touch "${NEXUS_DATA_DIR}"/migration
-fi
\ No newline at end of file
+fi
+
+# database backup script needs to be written to a file from string, because it is needed in the old container
+writeDatabaseBackupScriptToFile() {
+  export BACKUP_TASK_NAME="orientDatabaseBackup"
+  echo 'import org.sonatype.nexus.scheduling.TaskConfiguration; import org.sonatype.nexus.scheduling.TaskScheduler; def taskScheduler = container.lookup(TaskScheduler.class.getName()); TaskConfiguration config = taskScheduler.createTaskConfigurationInstance("db.backup"); config.setEnabled(true); config.setName("orientDatabaseBackup"); config.setString("location", "/opt/sonatype/nexus"); taskScheduler.submit(config);' > "${NEXUS_WORKDIR}/resources/nexusBackupOrientDBTask.groovy"
+}
+
+if [[ $FROM_VERSION == 3.70.2* ]] && [[ $TO_VERSION == 3.75.0* ]]; then
```

Comment:
Add comment to explain why only in this case. Wouldn't it be better to check for "jar file exist" && "db still OrientDB"

Reply:
Checking for migrator jar does not work as it is possible to download the jar in non-airgapped environments.
Checking for OrientDB is possible, but since the migration can only take place in 3.70.x I think it is not necessary

Reply:
But the fromVersion must be `3.70.2-5` or newer to have the migrator-jar included.
So I think that should be in there

Reply:
It is still possible to upgrade from another 3.70 version if the system has internet access